srcDLL/DllMain.cpp

```diff
@@ -40,34 +41,39 @@ AppEvents appEvents;
 BOOL WINAPI DllMain(HINSTANCE hInstance, DWORD dwReason, LPVOID /*reserved*/)
 {
 	// This will be called once the program is unpacked and running
-	if (dwReason == DLL_PROCESS_ATTACH) {
-		// Setup logging
-		SetLoggerError(&loggerDistributor);
-		SetLoggerMessage(&loggerFile);
-		SetLoggerDebug(&loggerDebug);
-
-		// Construct global objects
-		vols = std::make_unique<std::vector<std::string>>();
-		moduleLoader = std::make_unique<ModuleLoader>();
-
-		// Set load offset for Outpost2.exe module, used during memory patching
-		// If this fails, it's because Outpost2.exe is not loaded
-		// Failure means op2ext.dll was loaded by something else, such as a unit test
-		// For unit tests, just stay in memory, it's not an error if this fails
-		if (EnableOp2MemoryPatching()) {
-			// These hooks are needed to further bootstrap the rest of module loading
-			if (!InstallTAppEventHooks()) {
-				LogError("Failed to install initial TApp event hooks. Module loading and patching disabled.");
-				return FALSE;
+	if ((dwReason == DLL_PROCESS_ATTACH) && CommandOptionExists("OPU")) {
+		wchar_t filename[MAX_PATH] = L"";
+		GetModuleFileNameW(NULL, &filename[0], MAX_PATH);
+
+		if (fs::path(filename).filename() == "Outpost2.exe") {
```

Comment:
The winmm.dll shim (that loads op2ext.dll as an import) could potentially be loaded into any process, since it is a system DLL. This is just out of an abundance of caution.